File: Driver.java
//Driver.java will be utilized to test our code
//==============================================================================
/**
 * @file    Driver.java
 * @date    2/26/2018
 * @author  Cierra
 * @author  Ricardo
 *
 */
//==============================================================================
/**
 * @class Driver
 *
 * File utilized to test and simulate the implementation for CrossTheLine auto
 */
public class Driver
{
  public static void main(String[] args) throws InterruptedException
    {
      /// Functions shall be placed inside the while loop to simulate the
      /// number of iterations and the iteration speed it will encounter
      /// while running auto mode on RoboRio

      /// The iteration variable will count the number of iterations that occur
      /// in 15 seconds (750 at the 20ms per iteration rate of RoboRio)
      int iteration = 0;
      while (iteration<750)
      {
        /// Place function to be tested between here...
    	VictorSP leftmotor1 = new VictorSP(0);
    	VictorSP leftmotor2 = new VictorSP(1);
    	VictorSP leftmotor3 = new VictorSP(2);
    	
    	VictorSP rightmotor1 = new VictorSP(3);
    	VictorSP rightmotor2 = new VictorSP(4);
    	VictorSP rightmotor3 = new VictorSP(5);
    	
    	VictorSP[] leftmotors = new VictorSP[3];
    	leftmotors[0] = leftmotor1;
    	leftmotors[1] = leftmotor2;
    	leftmotors[2] = leftmotor3;
    	
    	VictorSP[] rightmotors = new VictorSP[3];
    	rightmotors[0] = rightmotor1;
    	rightmotors[1] = rightmotor2;
    	rightmotors[2] = rightmotor3;
    	
    	drive(3, leftmotors, rightmotors, 0.10, 0.10, "right", 250);
        /// ... and here!
        iteration++;
        /// Sleep the thread for 20ms to simulate iteration time of RoboRio
        Thread.sleep(20);
      }
    }
  
  
  static int counter = 0;

  // number_of_motors is the number of motors per side, not in total
  public static void drive(int number_of_motors, VictorSP[] left, VictorSP[] right, double leftspeed, double rightspeed, String reverse, int durationOfTravel) {
	  
	  if(counter<=durationOfTravel) {
		  
		  for (int i = 0; i < number_of_motors; i++) {
			  if(reverse.equalsIgnoreCase("left"))
				  leftspeed = -leftspeed;
			  left[i].set(leftspeed);
			  
			  if(reverse.equalsIgnoreCase("right"))
				  rightspeed = -rightspeed;
			  right[i].set(rightspeed);
			  
			  System.out.println("The robot is moving.");
		  }
		  
	  }
	  else {
		  
		  for (int i = 0; i < number_of_motors; i++) {
		  left[i].set(0.0);
		  right[i].set(0.0);
		  }
		  
		  System.out.println("The robot has stopped.");
		  
	  }
	  
	  counter++;
	  System.out.format("%d%n", counter);
	  
  }
  
  
  
}

File: CrossTheLine/Driver.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   Driver.cpp
 * Author: Cierra
 *
 * Created on March 1, 2018, 9:37 PM
 */

#include <iostream>
#include <cstdlib>
#include <thread>
#include <chrono>
#include "VictorSP.h"

using namespace std;
int counter = 0;
int iteration = 0;

void drive(int number_of_motors, VictorSP left[], VictorSP right[], double leftspeed, double rightspeed, bool reverseleft, bool reverseright, int durationOfTravel);

/*
 * 
 */



int main(int argc, char** argv) {

    while (iteration<500) {
        VictorSP leftmotor1 = VictorSP(0);
        VictorSP leftmotor2 = VictorSP(1);
        VictorSP leftmotor3 = VictorSP(2);
        VictorSP rightmotor1 = VictorSP(3);
        VictorSP rightmotor2 = VictorSP(4);
        VictorSP rightmotor3 = VictorSP(5);

        VictorSP leftmotors[3] = {leftmotor1, leftmotor2, leftmotor3};
        VictorSP rightmotors[3] = {rightmotor1, rightmotor2, rightmotor3};

        drive(3, leftmotors, rightmotors, 896, 675, false, true, 5);  
        
        iteration++;
        std::this_thread::sleep_for (std::chrono::milliseconds(20));
    }
   
    
}

// number_of_motors is the number of motors per side, not in total
void drive(int number_of_motors, VictorSP left[], VictorSP right[], double leftspeed, double rightspeed, bool reverseleft, bool reverseright, int durationOfTravel) {
    
    if(counter < durationOfTravel*50) {
        
        for (int i = 0; i < number_of_motors; i++) {
            
            if(reverseleft) {
                leftspeed = -leftspeed;
            }
            left[i].set(leftspeed);
        
            if(reverseright) {
                rightspeed = -rightspeed;
            }
            right[i].set(rightspeed);
            
            std::cout << "The robot is moving." << std::endl;
        }     
        
    }
    else {
        for (int i = 0; i < number_of_motors; i++) {
            left[i].set(0.0);
            right[i].set(0.0);
        }
        std::cout << "The robot has stopped." << std::endl;
    }    
        
    counter++;
    std::cout << counter << std::endl;
    
}